activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -458,7 +458,7 @@ def apply_join_dependency(eager_loading: group_values.empty?)
         join_dependency = construct_join_dependency(
           eager_load_values | includes_values, Arel::Nodes::OuterJoin
         )
-        relation = except(:includes, :eager_load, :preload).joins!(join_dependency)
```

Comment:
Did some git archaeology on why we stash the join in the `join_values` as opposed to `left_joins_values` (since the node is a Arel::Nodes::OuterJoin). It seems that this code was added before Rails had support for `LEFT OUTER JOIN` (added in https://github.com/rails/rails/commit/3f46ef1ddab)